app_dart/lib/src/model/firestore/pr_check_runs.dart

```diff
@@ -0,0 +1,111 @@
+// Copyright 2024 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:cocoon_service/src/service/logging.dart';
+import 'package:github/github.dart';
+
+import '../../service/firestore.dart';
+import 'package:googleapis/firestore/v1.dart' hide Status;
+
+/// Pairs the GitHub PR with the check runs associated with it.
+///
+/// The current webhook for check_runs does not include the PR. To reduce quota
+/// on GitHub, we need to be able to look up PRs based on check_run ids. Each
+/// document will be a collection of check_runs scheduled for a PR / sha.
+///
+/// The PR / SHA combo is not unique:
+///   - PR's will have multiple runs and we want this document to represent a
+///     snapshot.
+///   - SHAs can exist in multiple PRs.
+///
+/// Instead of generating some unique key (e.g. check_run+slug) and creating a
+/// large amount of documents, we will rely on querying the fields.
+///
+/// This document layout is currently:
+///  /projects/flutter-dashboard/databases/cocoon/prCheckRuns/
+///     document: <firebase unique id>
+///       pullRequest: json string
+///       slug: json string
+///       sha: string
+///       [*fields]: <string check_run id>: empty string
+class PrCheckRuns extends Document {
+  /// Firestore collection for the staging documents.
+  static const kCollectionId = 'prCheckRuns';
+  static const kPullRequestField = 'pull_request';
+  static const kSlugField = 'slug';
+  static const kShaField = 'sha';
+
+  /// Create [PrCheckRuns] from a Commit Document.
+  static PrCheckRuns fromDocument(Document prCheckRunsDoc) {
+    return PrCheckRuns()
+      ..fields = prCheckRunsDoc.fields!
+      ..name = prCheckRunsDoc.name!;
+  }
+
+  /// The json string of the pullrequest belonging to this document.
+  PullRequest get pullRequest => PullRequest.fromJson(json.decode(fields![kPullRequestField]!.stringValue!));
+
+  /// The head sha at the time this document was created for testing.
+  String get sha => fields![kShaField]!.stringValue!;
+
+  RepositorySlug get slug => RepositorySlug.fromJson(json.decode(fields![kSlugField]!.stringValue!));
+
+  /// Initializes a new document for the list of check_runs in Firestore so we can find it later.
+  ///
+  /// The list of tasks will be written as fields of a document with additional fields for tracking the total
+  /// number of tasks, remaining count. It is required to include [checkRunGuard] as a json encoded [CheckRun] as this
+  /// will be used to unlock any check runs blocking progress.
+  ///
+  /// Returns the created document or throws an error.
+  static Future<Document> initializeDocument({
+    required FirestoreService firestoreService,
+    required PullRequest pullRequest,
+    required List<CheckRun> checks,
```

Comment:
We should maybe combine this PR with your previous one that ignores ci yaml and merge guard, and make sure that we do not add those to this list.

Reply:
I don't - because those are created in Scheduler and the luci checks are created in luci.  Even if they were recorded, it doesn't matter since this is just a <check run> -> <pr> helper.